client/src/components/DataFiles/DataFilesToolbar/DataFilesToolbar.test.jsx

```diff
@@ -233,4 +240,114 @@ describe('DataFilesToolbar', () => {
+      name: 'test.txt',
+      type: 'file',
+      length: tooBigFileSize,
+      path: '/test.txt',
+    };
+    // Create the store
+    const { getByText } = renderComponent(
+      <DataFilesToolbar scheme="private" api="tapis" />,
+      mockStore({
+        workbench: {
+          config: {
+            extract: '',
+            compress: '',
+            trashPath: '.Trash',
+          },
+        },
+        files: {
+          params: {
+            FilesListing: {
+              system: 'frontera.home.username',
+              path: 'home/username',
+              scheme: 'private',
+            },
+          },
+          listing: { FilesListing: [testFile] },
+          selected: { FilesListing: [0] },
+          operationStatus: { trash: false },
+        },
+        systems: systemsFixture,
+        projects: { metadata: [] },
+        authenticatedUser: { user: { username: 'testuser' } },
+      })
+    );
+    // Click on the download button to try and download the file
+    fireEvent.click(getByText('Download'));
+    // Wait for the Large Download Modal
+    await waitFor(() => screen.queryByText('Large Download'));
+    // Assign the Large Download Modal to a variable
+    const testModal = screen.queryByText('Large Download');
+    // Test for the Large Download Modal
+    expect(testModal).toBeDefined();
+  });
+
+  // Test that allows downloads of files less than 2 GB
+  it('allows direct file downloads when the file size is below 2 GB', () => {
+    // Mock the dispatch action
+    const mockDispatch = vi.fn();
+    // Create a test file whose size is less than 2 GB
+    const testFileSize = 1 * 1024 * 1024 * 1024;
+    const testFile = {
+      name: 'test.txt',
+      type: 'file',
+      length: testFileSize,
+      path: '/test.txt',
+      id: 123,
+    };
+    // Create a spy that watches for the dispatch call
+    vi.spyOn(require('react-redux'), 'useDispatch').mockReturnValue(
+      mockDispatch
+    );
+    // Create the store
+    const { getByText } = renderComponent(
+      <DataFilesToolbar scheme="private" api="tapis" />,
+      mockStore({
+        workbench: {
+          config: {
+            extract: '',
+            compress: '',
+            trashPath: '.Trash',
+          },
+        },
+        files: {
+          params: {
+            FilesListing: {
+              system: 'frontera.home.username',
+              path: 'home/username',
+              scheme: 'private',
+            },
+          },
+          listing: { FilesListing: [testFile] },
+          selected: { FilesListing: [0] },
+          operationStatus: { trash: false },
+        },
+        systems: systemsFixture,
+        projects: { metadata: [] },
+        authenticatedUser: { user: { username: 'testuser' } },
+      })
+    );
+    // Click on the download button to try and download the file
+    fireEvent.click(getByText('Download'));
+    // Test for the dispatch call
+    expect(mockDispatch).toHaveBeenCalledWith({
```

Comment:
This expectation isn't being met in this test, and I'm not sure why. I noticed that you are selecting the `Download` button differently than the other tests are, it could be that the test is firing a click on the wrong thing. Or, it could be that the modal is sending the toggle dispatch before the download dispatch.